packages/gatsby/src/utils/validate-page-component.ts

```diff
@@ -4,6 +4,9 @@ import { IGatsbyPage } from "../redux/types"
 
 const validationCache = new Set<string>()
 
+const sourceExtensionsArray = [`jsx`, `tsx`, `js`, `ts`]
```

Comment:
I think these need to have the dot, because the value returned by `path.extname` includes it

Reply:
Both versions seem to work and I'm not sure why. But happy to add the `.` for consistency.